src/main/java/com/frameworkium/core/ui/tests/BaseTest.java

```diff
@@ -0,0 +1,258 @@
+
+    private static final long DEFAULT_TIMEOUT_SECONDS = 10;
+
+    private static String userAgent;
+
+    private static ThreadLocal<Boolean> requiresReset;
+    private static ThreadLocal<ScreenshotCapture> capture;
+    private static ThreadLocal<Driver> driver;
+    private static ThreadLocal<Wait<WebDriver>> wait;
+    private static List<Driver> activeDrivers =
+            Collections.synchronizedList(new ArrayList<>());
+    private static Logger logger = LogManager.getLogger(BaseTest.class);
+
+    /**
+     * Method which runs first upon running a test, it will do the following:
+     * - Retrieve the desired {@link Driver} and initialise the {@link WebDriver}
+     * - Initialise the {@link Wait}
+     * - Initialise whether the browser needs resetting
+     * - Initialise the {@link ScreenshotCapture}
+     */
+    @BeforeSuite(alwaysRun = true)
+    public static void instantiateDriverObject() {
+        driver = ThreadLocal.withInitial(() -> {
+            Driver newDriver =
+                    new DriverSetup().returnDesiredDriverType();
+            newDriver.instantiate();
+            activeDrivers.add(newDriver);
+            return newDriver;
+        });
+        wait = ThreadLocal.withInitial(BaseTest::newDefaultWait);
+        requiresReset = ThreadLocal.withInitial(() -> Boolean.FALSE);
+        capture = ThreadLocal.withInitial(() -> null);
+    }
+
+    public static Wait<WebDriver> newDefaultWait() {
+        return newWaitWithTimeout(DEFAULT_TIMEOUT_SECONDS);
+    }
+
+    public static Wait<WebDriver> newWaitWithTimeout(long timeout) {
+        return new FluentWait<>(getDriver().getWrappedDriver())
+                .withTimeout(timeout, TimeUnit.SECONDS)
+                .ignoring(org.openqa.selenium.NoSuchElementException.class)
+                .ignoring(StaleElementReferenceException.class);
+    }
+
+    /**
+     * Returns the {@link WebDriverWrapper} instance for the requesting thread
+     *
+     * @return - WebDriver object
+     */
+    public static WebDriverWrapper getDriver() {
+        return driver.get().getDriver();
+    }
+
+    /**
+     * The methods which configure the browser once a test runs
+     * <ul>
+     * <li>Maximises browser based on the driver type</li>
+     * <li>Initialises screenshot capture if needed</li>
+     * <li>Resets the browser if another test ran prior</li>
+     * <li>Sets the user agent of the browser</li>
+     * </ul>
+     *
+     * @param testMethod - The test method name of the test
+     */
+    @BeforeMethod(alwaysRun = true)
+    public static void configureBrowserBeforeTest(Method testMethod) {
+        try {
+            configureDriverBasedOnParams();
+            initialiseNewScreenshotCapture(testMethod);
+        } catch (Exception e) {
+            logger.error("Failed to configure browser.", e);
+            throw new RuntimeException("Failed to configure browser.", e);
+        }
+    }
+
+    /** TODO: Should be moved inside the Driver object */
```

Comment:
ForceVisible - agreed, but it's something that I've hit a few times in one of my projects. The main use case so far: checkboxes obscured by prettifying widgets (eg an on/off slider) - where there's no way of telling whether the state of it (eg if the slider is set to on or off) without interrogating the underlying 'input'. This requires bringing it out into the open, irritatingly.

It's only intended as a last resort, but really neatens the code on pages where it's required.